--- libphonenumber-8.13.8/cpp/CMakeLists.txt.orig	2023-02-26 10:49:15.997683242 +0100
+++ libphonenumber-8.13.8/cpp/CMakeLists.txt	2023-02-26 17:16:58.674991835 +0100
@@ -106,7 +106,7 @@ if (USE_ALTERNATE_FORMATS)
 endif ()
 
 # Find all the required libraries and programs.
-find_package(absl)
+find_package(absl REQUIRED COMPONENTS node_hash_set strings synchronization)
 
 if(NOT absl_FOUND)
   # Overide abseil install rules for subprojects
