--- libphonenumber-8.13.8/cpp/CMakeLists.txt.orig	2023-01-06 01:42:02.871437832 +0100
+++ libphonenumber-8.13.8/cpp/CMakeLists.txt	2023-01-06 01:49:02.681479463 +0100
@@ -599,7 +599,11 @@ if(BUILD_TESTING)
   # libraries are built properly.
   if (BUILD_GEOCODER)
     add_executable (geocoding_test_program "test/phonenumbers/geocoding/geocoding_test_program.cc")
+   if (${BUILD_STATIC_LIB} STREQUAL "ON")
     target_link_libraries (geocoding_test_program geocoding phonenumber)
+   else ()
+    target_link_libraries (geocoding_test_program geocoding-shared phonenumber-shared)
+   endif ()
   endif ()
 endif()
 
