--- libphonenumber-8.12.51/cpp/CMakeLists.txt.orig	2023-01-06 01:42:02.871437832 +0100
+++ libphonenumber-8.12.51/cpp/CMakeLists.txt	2023-01-06 01:49:02.681479463 +0100
@@ -498,7 +498,9 @@
   list (APPEND GEOCODER_DEPS ${COMMON_DEPS})
   # Note that the subset of base/ on which the geocoder relies is implemented
   # on top of Boost header-only libraries (e.g. scoped_ptr.hpp).
+  if (${BUILD_STATIC_LIB} STREQUAL "ON")
   target_link_libraries (geocoding ${LIBRARY_DEPS})
+  endif()
   target_link_libraries (geocoding-shared phonenumber-shared ${LIBRARY_DEPS})
 endif ()
 
@@ -665,7 +667,9 @@
 endif ()
 
 if (${BUILD_GEOCODER} STREQUAL "ON")
+  if (${BUILD_STATIC_LIB} STREQUAL "ON")
   install (TARGETS geocoding LIBRARY DESTINATION ${LIBDIR} ARCHIVE DESTINATION ${LIBDIR})
+  endif()
   install (TARGETS geocoding-shared LIBRARY DESTINATION ${LIBDIR} ARCHIVE
            DESTINATION ${LIBDIR})
 endif ()
@@ -677,7 +681,11 @@
     geocoding_test_program
     "test/phonenumbers/geocoding/geocoding_test_program.cc"
   )
+  if (${BUILD_STATIC_LIB} STREQUAL "ON")
   target_link_libraries (geocoding_test_program geocoding phonenumber)
+  else ()
+  target_link_libraries (geocoding_test_program geocoding-shared phonenumber-shared)
+  endif ()
 endif ()
 
 # Build an RPM
